### PR TITLE
fix(wallet): handle cancelled offer state

### LINKED ISSUE #4243: Wallet UI: "Cancelled" offers show buggy state
## Describe the bug

After exiting an offer, the backend changes its status to "cancel". The ui does not handle this state, instead showing an empty status badge on the offer and no option to dismiss it.

## To Reproduce
1. Make a bid on a card in dapp-card-store
2. After approving the offer in the wallet, click the "exit" button
3. Observe the state change to show like the screenshot below

## Expected behavior
The badge should show "cancelled" and the offer should be greyed out and dismissible like a completed offer.

## Screenshots
image

### PR DESCRIPTION
**Pull request by samsiegart**
Fixes: https://github.com/Agoric/agoric-sdk/issues/4243
Screenshot:
image

### CODE REVIEW

**Comment by samsiegart** [comment]
I wonder if it makes sense to show what assets were actually transferred vs returned inside the "completed" offer. I noticed that even if you were outbid on a card the offer shows "completed" but you didn't actually lose your give/get your want.

**Comment by michaelfig** [comment]
> I wonder if it makes sense to show what assets were actually transferred vs returned inside the "completed" offer. I noticed that even if you were outbid on a card the offer shows "completed" but you didn't actually lose your give/get your want.

This is a good idea, maybe a "refunded" state.  Zoe currently doesn't do anything to distinguish this result, so it would take some work on the Zoe service to be able to report it.  We could instead hack together something in the wallet backend to heuristically determine that it was a refund (not my preference).

@erights, thoughts?

File: packages/wallet/ui/src/components/Offer.jsx [hunk 1]
```diff
@@ -194,17 +196,15 @@ const OfferWithoutContext = ({
     </div>
   );
 
+  const isOfferCompleted =
+    status === 'accept' ||
+    status === 'decline' ||
+    status === 'complete' ||
+    status === 'rejected' ||
+    status === 'cancel';
+
```
Comment (michaelfig) [inline packages/wallet/ui/src/components/Offer.jsx]:
```suggestion
  const isOfferCompleted = ['accept', 'decline', 'complete', 'rejected', 'cancel'].includes(status);
```

**Review by michaelfig — APPROVED** [review]
LGTM!  Just one suggestion to make a conditional terser.

File: packages/wallet/ui/src/components/Offer.jsx (hunk 1 above)
Comment (erights) [reply packages/wallet/ui/src/components/Offer.jsx]: @samsiegart @michaelfig 
Where does this `status` string come from? Where are its possible values defined?

Comment (samsiegart) [reply packages/wallet/ui/src/components/Offer.jsx]: I'm not sure if we have any good documentation on that. The implementation is in https://github.com/Agoric/agoric-sdk/blob/master/packages/wallet/api/src/lib-wallet.js (search for "status:")

**Comment by samsiegart** [comment]
Will merge to fix #4243, we can continue the discussion here or open an issue if need be

**Pull request merged**